### PR TITLE
feat(cli): Make versioned quickstart downloads

### PR DESCRIPTION
**Pull request by szalai1**
Add --unstable flag to pick up latest from head, but defaults to stable behaviour

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/cli/docker_cli.py [hunk 1]
```diff
@@ -670,6 +718,12 @@ def quickstart(
     auth_resources_folder = Path(DATAHUB_ROOT_FOLDER) / "plugins/auth/resources"
     os.makedirs(auth_resources_folder, exist_ok=True)
 
+    stable = not unstable
+    if stable:
```
Comment (pedro93) [inline metadata-ingestion/src/datahub/cli/docker_cli.py:722]: If unstable, where does the code pick up the compose files from?

Comment (pedro93) [reply metadata-ingestion/src/datahub/cli/docker_cli.py:722]: Can we still use DOCKER_COMPOSE_BASE env var to specify where to get the compose files from?
We use it in a few places:
- https://github.com/datahub-project/datahub/blob/7cb3d7209cb5219dd4ae149809287fb6e0725a94/smoke-test/run-quickstart.sh#L17
- https://github.com/datahub-project/datahub/blob/7cb3d7209cb5219dd4ae149809287fb6e0725a94/docker/build.gradle#L25
- https://github.com/datahub-project/datahub/blob/7cb3d7209cb5219dd4ae149809287fb6e0725a94/metadata-integration/java/spark-lineage/spark-smoke-test/smoke.sh#L43

File: metadata-ingestion/src/datahub/cli/docker_cli.py [hunk 2]
```diff
@@ -803,28 +858,44 @@ def quickstart(
     )
 
 
+def get_docker_compose_base_url(version_tag: Optional[str]) -> str:
+    if os.environ.get("DOCKER_COMPOSE_BASE"):
```
Comment (szalai1) [inline metadata-ingestion/src/datahub/cli/docker_cli.py:862]: @pedro93 you can use the `DOCKER_COMPOSE_BASE`

File: metadata-ingestion/src/datahub/cli/docker_cli.py [hunk 3]
```diff
@@ -803,28 +858,44 @@ def quickstart(
     )
 
 
+def get_docker_compose_base_url(version_tag: Optional[str]) -> str:
```
Comment (szalai1) [inline metadata-ingestion/src/datahub/cli/docker_cli.py:861]: @pedro93 
>If unstable, where does the code pick up the compose files from?
This method decided where will the files picked up from.

File: metadata-ingestion/src/datahub/cli/docker_cli.py (hunk 2 above)
Comment (pedro93) [reply metadata-ingestion/src/datahub/cli/docker_cli.py:862]: Missed this, was still going through the PR. Thanks!

**Review by pedro93 — APPROVED** [review]
LGTM assuming CI runs.

**Review by david-leifker — CHANGES_REQUESTED** [review]
Please hold on this until we can sync

File: metadata-ingestion/src/datahub/cli/docker_cli.py (hunk 3 above)
Comment (pedro93) [reply metadata-ingestion/src/datahub/cli/docker_cli.py:858]: What if we want an unstable version from a different compose base path? I.e: To test HEAD images with compose PR changes?

**Comment by jjoyce0510** [comment]
Pending an update here.

**Pull request closed without merging**